src/Collection/TableCollection.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Quidco\DbSampler\Collection;
+
+use Quidco\DbSampler\Configuration\MigrationConfiguration;
+use Quidco\DbSampler\ReferenceStore;
+use Quidco\DbSampler\SamplerInterface;
+use Quidco\DbSampler\SamplerMap\SamplerMap;
+
+class TableCollection
+{
+    private $tables = [];
+
+    /**
+     * @var ReferenceStore
+     */
+    private $referenceStore;
+    /**
+     * @var array
+     */
+    private $rawTables;
+
+    private function __construct(array $rawTables)
+    {
+        $this->referenceStore = new ReferenceStore();
+        $this->rawTables = $rawTables;
+    }
+
+    public static function fromConfig(MigrationConfiguration $configuration): self
+    {
+        return new self((array)$configuration->getTables());
+    }
+
+    public function getTables(): array
```

Comment:
Can we avoid repeating `Collection` in the naming here?
Something like:
`namespace Quidco\DbSampler\Tables::getTableSamplers()` 🤔

Reply:
I'm hesitant to call the method `getTableSamplers` because I don't think the sampler should be a concern of this class, which is just a list of the configured tables. At the moment there doesn't seem to be another place to put it (without extra work), but once things are properly decoupled we should see a natural home emerging for where to build the sampler. The `Collection` name space is more temporary than fixed (again for the purpose of a long-game refactor), and `Table` / `View`, etc _might_ make more sense

Reply:
thinking about it, _maybe_ `DatabaseSchema` might be more descriptive as a namespace for the `Table` and `View` collection classes

Reply:
I made this change in #b8c9ca4b